sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -56,12 +55,16 @@ abstract class RuleExecutor[TreeType <: TreeNode[_]] extends Logging {
   case class FixedPoint(maxIterations: Int) extends Strategy
 
   /** A batch of rules. */
-  protected case class Batch(name: String, strategy: Strategy, rules: Rule[TreeType]*)
+  case class Batch[TreeType <: TreeNode[_]]
```

Comment:
Got it, i moved Batch and Strategy to the rules package, and made Once and FixedPoint classes of the companion object of Strategy.